src/collective/glossary/browser/views.py

```diff
@@ -135,11 +136,11 @@ class JsonView(BrowserView):
     def __init__(self, context, request):
         self.context = context
         self.request = request
-        self.items = self.prepare_items()
 
     @ram.cache(_catalog_counter_cachekey)
-    def prepare_items(self):
-        """Get all itens and prepare in the desired format"""
+    def get_json_entries(self):
+        """Get all itens and prepare in the desired format.
+        Note: do not name it get_entries, otherwise caching is broken. """
```

Comment:
Using the same function name will result in Mixing the ram cache from JsonView and GlossaryView which will not work.